partiql-eval/src/main/java/org/partiql/eval/value/PQLValue.java

```diff
@@ -0,0 +1,546 @@
+package org.partiql.eval.value;
+
+import kotlin.NotImplementedError;
+import kotlin.Pair;
+import org.jetbrains.annotations.NotNull;
+import org.partiql.value.PartiQL;
+import org.partiql.value.PartiQLValue;
+import org.partiql.value.PartiQLValueType;
+import org.partiql.value.datetime.Date;
+import org.partiql.value.datetime.Time;
+import org.partiql.value.datetime.Timestamp;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.BitSet;
+import java.util.Iterator;
+import java.util.Objects;
+
+/**
+ * This is a representation of a value in PartiQL's type system. The intention of this modeling is to provide a layer of
+ * indirection between PartiQL's type semantics and Java's type semantics.
+ *
+ * <p></p>
+ * INTERNAL NOTES:
+ * <p></p>
+ * This is intended to completely replace {@link org.partiql.value.PartiQLValue} in the future. As it stands, this
+ * implementation will initially be used solely for the evaluator. However, the scope of this can be expanded by copying
+ * and pasting its contents to completely replace {@link org.partiql.value.PartiQLValue}.
+ * <p></p>
+ * There are some pre-requisites to actually replacing {@link PartiQLValue} including, but not limited to, ...:
+ * - The comparator for ordering and aggregations
+ * - Adding support for annotations
+ */
+public interface PQLValue extends Iterable<PQLValue> {
```

Comment:
I don't think it should be public if it's not a replacement. I also thought we discussed them solving different problems?

We shouldn't introduce another duplication of something existing when PartiQLValue is _still_ under experimental. We either replace it completely or change it however we need to.